src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -700,6 +700,14 @@ internal void Parse(string[] args)
                 throw new InvalidOperationException("This instance has already been used. Create a new instance.");
             }
 
+            for (int i = 0; i < args.Length; i++)
+            {
+                if (args[i] is null)
+                {
+                    throw new ArgumentNullException("The 'args' array contains a null element.");
```

Comment:
The message should be put in .resx file. Make the message to "The specified arguments should not contain null elements.". Also, please pass in the parameter name to `ArgumentNullException`.

Reply:
`nameof(args)`

Reply:
No exceptions in the file use Resx file.
Please confirm using Resx file in the case.

Reply:
I think you need to differentiate an exception thrown due to user action from an exception that's thrown due to internal error only.
The `InvalidOperationException` thrown when `_dirty == true` can happen only when there is an internal error -- the parser gets reused somehow. It shouldn't happen and is not user facing.
The `ArgumentNullException` thrown here only because a user action -- the user pass in invalid args. It's user facing.

Reply:
I see your point. The issue could be raised (1) only in host application, (2) only if the host application developer constructs `args` with null element. So it is not "end user action" but a bug in the host application that is "internal error" case too. Thoughts?

Reply:
Anyone using a PowerShell public API is an end user. An internal error is an error that cannot be triggered by an end user.
I hope this makes it more clear to you.

Reply:
Thanks for clarify! 

Done.